crates/wash-lib/src/cli/registry.rs

```diff
@@ -41,8 +41,8 @@ pub enum RegistryCommand {
 #[derive(Parser, Debug, Clone)]
 pub struct RegistryPullCommand {
     /// URL of artifact
-    #[clap(name = "url")]
-    pub url: String,
+    #[clap(name = "url", env = "WASH_REG_URL")]
+    pub url: Option<String>,
```

Comment:
Hmnnn I think the ENV variable might be a bit weirdly named. This URL is normally the URL of *the artifact*, which happens to be in a registry.  Maybe this should be `WASH_ARTIFACT_URL` ?

As written, I'd think of a registry URL to be `ghcr.io/user`, and an artifact URL to be `ghcr.io/user/project:0.1.0`... does that match with how you're thinking about it?

Reply:
so, the distinction did cross my mind. Though, looking at the related issue (and its discussion) and the related code, there is no apparent distinction between a registry url and artifact url (ie, a single registry providing multiple artifact repositories). if the distinction does matter (and it probably should), then there needs to be more refinement on the issue to address how we want to handle it across all config options (cli, env, wasmcloud file, custom and default credentials file).

Reply:
Hmnnn, so when I look around the codebase I mostly see uses of URLs in the context of artifacts -- the places I do see a "registry url" (like `oci.rs`), It's intended to be the URL of the registry itself (not an individual manifest or artifact). 

What really made me think that the name might be more geared was two things:

- Doc comment above there that mentions artifacts (my first thought was that maybe we need to update that doc comment, but on second thought it seems like an artifact URL)
- How we use [the `url` parameter on `RegistryPushCommand`](https://github.com/wasmCloud/wasmCloud/blob/main/crates/wash-cli/src/common/registry_cmd.rs#L33) -- we call it an `artifact_url` there.

I'm willing to be flexible here though, if you feel like `WASH_REG_URL` is the right name for this. It's not like we can't change it later, and I'd love to get your change merged sooner rather than later!

Reply:
on the decision on how to move forward with this, i will defer to @thomastaylor312 since he is the one who originally created the issue #860. the solution i proposed (see discussion on the issue itself) isn't a good fit for having a common configured registry to which multiple artifact can be pushed/pulled.  so, we can either move forward with this for now or this PR is put on ice until an adequate solution is devised.

Reply:
Let's move forward with this for now! Given that it wasn't used before I think we've got some space -- I'll leave this open just in case Taylor wants to chime in.

Reply:
Taylor is out for the next week, but I'll comment on his behalf. Up to now, this url was meant to be a specific artifact URL, e.g. `wasmcloud.azurecr.io/echo:0.3.8`, not just a registry url (e.g. `wasmcloud.azurecr.io`). 

Currently, I can do:

```
wash reg pull/push wasmcloud.azurecr.io/echo:0.3.8
```

With your changes, we allow users to set `registry` config in the project file. So that _might_ allow usage like:
```
wash reg pull/push echo:0.3.8
```

However, since the registry config is optional, we won't always know what the registry URL is. Another option is to change the command structure to be something like this:
```
wash reg pull/push --registry wasmcloud.azurecr.io echo:0.3.8
```

And then in cases where I've set up `registry` config in my `wasmcloud.toml`, I could omit the flag and just do:

```
wash reg pull/push echo:0.3.8
```